pallets/aleph/src/mock.rs

```diff
@@ -173,18 +174,26 @@ pub fn new_test_ext(authorities: &[(u64, u64)]) -> sp_io::TestExternalities {
     t.into()
 }
 
-pub(crate) fn run_session(n: u64) {
-    while System::block_number() < n {
+pub(crate) fn run_session(n: u32) {
+    for i in Session::current_index()..n {
         Session::on_finalize(System::block_number());
         Aleph::on_finalize(System::block_number());
         System::on_finalize(System::block_number());
```

Comment:
not sure how this is relevant, but as far as I understand, the hook `on_finalize` is called on pallets in the reversed order than the one defined in `bin/runtime/src/lib.rs`;
however, we ordered Aleph after Session, so in order to mock the real behavior we should keep the true succession

Reply:
Good catch, changed.

Reply:
almost :) the `on_initialize` block below is out-of-order :stuck_out_tongue: